core/dbt/cli/context.py

```diff
@@ -0,0 +1,39 @@
+from click import Context, Group, Command
+from click.exceptions import NoSuchOption, UsageError
+from dbt.config.runtime import load_project, load_profile
+from dbt.cli.flags import Flags
+import sys
+
+
+class DBTUsageException(Exception):
+    pass
+
+
+class DBTContext(Context):
+    def __init__(self, command: Command, **kwargs) -> None:
```

Comment:
Trying to understand what different behavior will I get by passing in command `cli(group)` with all args vs `run` with all args

Reply:
So far this only really works for `cli`. The more I think it, the more it makes sense that only `the cli.context_class` should be `DBTContext` (as opposed to the subcommands) given all the top-level configuration and initialization it's doing.